src/main/webapp/app/entities/oauth-client/oauth-client-dialog.component.ts

```diff
@@ -0,0 +1,186 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Response } from '@angular/http';
+
+import { NgbActiveModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { EventManager, AlertService, JhiLanguageService } from 'ng-jhipster';
+
+import { OAuthClient } from './oauth-client.model';
+import { OAuthClientService } from './oauth-client.service';
+import { OAuthClientPopupService } from './oauth-client-popup.service';
+
+@Component({
+    selector: 'jhi-oauth-client-dialog',
+    templateUrl: './oauth-client-dialog.component.html'
+})
+export class OAuthClientDialogComponent implements OnInit  {
+
+    client: OAuthClient;
+    isSaving: boolean;
+    authorities: any[];
+    grantTypes: any[];
+    scopeList: string;
+    resourcesList: string;
+    autoApproveScopeList: string;
+    availableGrants: string[];
+    dynamicRegistration: boolean;
+    dynamicRegistrationKey: string;
+    newClient: boolean;
+    protectedClient: boolean;
+
+    constructor(
+        public activeModal: NgbActiveModal,
+        private jhiLanguageService: JhiLanguageService,
+        private alertService: AlertService,
+        private oauthClientSerivce: OAuthClientService,
+        private eventManager: EventManager
+    ) {
+        this.jhiLanguageService.addLocation('oauthClient');
+        this.availableGrants = ['authorization_code', 'implicit', 'client_credentials', 'refresh_token', 'password'];
+    }
+
+    ngOnInit() {
+        this.isSaving = false;
+        // transform array of authorized grant types to array of objects with name and authorized property, so we can bind a checkbox to the authorized property
+        this.grantTypes = [];
+        this.availableGrants.forEach((grant) => {
+            this.grantTypes.push({"name": grant, "authorized": this.client.authorizedGrantTypes.indexOf(grant) > -1});
+        });
+        // transform array of scopes to a comma seperated string so we can bind a single text box to it
+        this.scopeList = this.client.scope.join(", ");
+        // transform array of resources to a comma seperated string so we can bind a single text box to it
+        this.resourcesList = this.client.resourceIds.join(", ");
+        // transform array of auto-approve scopes to a comma seperated string so we can bind a single text box to it
+        this.autoApproveScopeList = this.client.autoApproveScopes.join(", ");
```

Comment:
Add ` this.autoApproveScopeList = this.client.autoApproveScopes? this.client.autoApproveScopes.join(", "):"" ;`
The UI crashes when autoApproveScopes is null.